[MM-35653] Data Retention redux tests

BenCookie95 opened the pull request:
#### Summary
Redux tests for data retention

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35653

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates)
* API additions—new endpoint, new response fields, or newly accepted request parameters
* Database changes (any)
* Websocket additions or changes
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating)
* New features and improvements, including behavioural changes, UI changes and CLI changes
* Bug fixes and fixes of previous known issues
* Deprecation warnings, breaking changes, or compatibility notes

If no release notes are required write NONE. Use past-tense. Newlines are stripped.

Examples:

```release-note
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```release-note
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```release-note
NONE
```
-->
```release-note
NONE
```

shazm reviewed (commented):
I think I'm probably out of my depth here with the redux updates / tests, but whats the difference `bindClientFunc` vs an `async dispatch / getState`? It seems more readable to me, and do we have a descriptive issue / ticket somewhere for removing the `include_total_count`? https://mattermost.atlassian.net/browse/MM-35653 doesn't tell me much 😅

BenCookie95 commented:
> I think I'm probably out of my depth here with the redux updates / tests, but whats the difference `bindClientFunc` vs an `async dispatch / getState`? It seems more readable to me, and do we have a descriptive issue / ticket somewhere for removing the `include_total_count`? https://mattermost.atlassian.net/browse/MM-35653 doesn't tell me much 😅

the bindClientFunc was not returning data like I wanted it to  so I wrote the async dispatch myself. 

I removed `include_total_count` because it wasn't a parameter on that endpoint and it wasn't being used anywhere. There is no ticket for it, I just cleaned it up in this PR

furqanmlk reviewed (approved):
thanks @BenCookie95 

All tests passed

![image](https://user-images.githubusercontent.com/6909890/120367848-5d13b400-c2df-11eb-8079-7165548a4bd2.png)
